pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -222,68 +221,71 @@ def set_config(self, cfg: SimConfig) -> None:
+            prob = self.config.dephasing_prob
             n = self._size
             if prob > 0.1 and n > 1:
                 warnings.warn(
                     "The dephasing model is a first-order approximation in the"
-                    f" dephasing probability. p = {2*prob} is too large for "
+                    f" dephasing probability. p = {prob} is too large for "
                     "realistic results.",
                     stacklevel=2,
                 )
-            k = np.sqrt(prob * (1 - prob) ** (n - 1))
-
+            if (
+                self.basis_name == "all" and self.dim == 3
+            ):  # three-level system
+                k = np.sqrt(prob * (1 - prob) ** (n - 1))
+                m_0 = np.sqrt(1 - prob) * qutip.qeye(3)
+                for i in range(3):
+                    ket_ = qutip.ket(str(i), 3)
+                    kraus_ops.append(k * ket_ * ket_.dag())
+                kraus_ops.append(m_0)
+            else:  # two-level system
+                prob = prob / 2
+                k = np.sqrt(prob * (1 - prob) ** (n - 1))
+                kraus_ops.append(k * qutip.sigmaz())
             self._collapse_ops += [
                 np.sqrt((1 - prob) ** n)
                 * qutip.tensor([self.op_matrix["I"] for _ in range(n)])
             ]
-            kraus_ops.append(k * qutip.sigmaz())
 
         if "depolarizing" in self.config.noise:
-            if self.basis_name == "digital" or self.basis_name == "all":
-                # Go back to previous config
-                self.set_config(prev_config)
-                raise NotImplementedError(
-                    "Cannot include depolarizing "
-                    + "noise in digital- or all-basis."
-                )
-            # Probability of error occurrence
-
-            prob = self.config.depolarizing_prob / 4
+            prob = self.config.depolarizing_prob
             n = self._size
             if prob > 0.1 and n > 1:
                 warnings.warn(
                     "The depolarizing model is a first-order approximation"
-                    f" in the depolarizing probability. p = {4*prob}"
+                    f" in the depolarizing probability. p = {prob}"
                     " is too large for realistic results.",
                     stacklevel=2,
                 )
+            if (
+                self.basis_name == "all" and self.dim == 3
+            ):  # three-level system
+                prob = prob / 9
+
+                k = np.sqrt((prob) * (1 - 8 * prob) ** (n - 1))
+                m_0 = np.sqrt(1 - prob) * qutip.qeye(3)
+                for i in range(3):
+                    ket_ = qutip.ket(str(i), 3)
+                    kraus_ops.append(k * ket_ * ket_.dag())
+                kraus_ops.append(m_0)
+                self._collapse_ops += [
+                    np.sqrt((1 - 8 * prob) ** n)
+                    * qutip.tensor([self.op_matrix["I"] for _ in range(n)])
+                ]
```

Comment:
Okay, so on this one if you divide by 3 you can use the same line for adding to `_collapse_ops` as is being used for the qubit case. The rest should be like this:
```suggestion
                prob = prob / 3

                k = np.sqrt((prob) * (1 - 3 * prob) ** (n - 1))
                for i in range(3):
                    ket_ = qutip.basis(3, 1)
                    kraus_ops.append(k * ket_ * ket_.dag())
```